server/services/build-service.js

```diff
@@ -24,6 +25,19 @@ module.exports = ({ strapi }) => ({
 			if (settings.body) {
 				requestConfig.data = settings.body;
 			}
+
+			if (settings.secret) {
+				const { hash, headerKey, secretKey } = settings.secret;
+				const normalizedHash = hash.toLowerCase();
+				requestConfig.headers = {
+					...(requestConfig.headers || {}),
+					[headerKey]: `${normalizedHash}=${crypto
+						.createHmac(normalizedHash, secretKey)
+						.update(JSON.stringify(requestConfig.data))
+						.digest('hex')}`,
```

Comment:
This should be in a utility function, along with the hash normalization on line 31.

Also what is the reason for the hash=digest format? Do we not set the value for the header to be the returned hash?

Update key/variable names per above comments.